src/NUnitFramework/framework/Internal/IgnoredTestCaseData.cs

```diff
@@ -0,0 +1,95 @@
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// The IgnoredTestCaseData class represents a ignored TestCaseData. It adds
+    /// the ability to set an ignore until date on a test case.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public sealed class IgnoredTestCaseData : TestCaseData
+    {
+        #region Instance Fields
+
+        /// <summary>
+        /// The previous RunState
+        /// </summary>
+        private RunState _prevRunState;
+
+        #endregion
+
+        #region Constructors
+
+        internal IgnoredTestCaseData(TestCaseData data, RunState prevRunState)
+        {
+            this.Arguments = data.Arguments;
+            this.ArgDisplayNames = data.ArgDisplayNames;
+            this.ExpectedResult = data.ExpectedResult;
+            this.HasExpectedResult = data.HasExpectedResult;
+            this.OriginalArguments = data.OriginalArguments;
+            this.Properties = data.Properties;
+            this.RunState = data.RunState;
+            this.TestName = data.TestName;
+            this._prevRunState = prevRunState;
+        }
+
+        #endregion
+
+        #region Fluent Instance Modifiers
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="datetime">The ignore until date</param>
+        /// <returns>A modified TestCaseData.</returns>
+        public TestCaseData Until(DateTimeOffset datetime)
+        {
+            if (this._prevRunState != RunState.NotRunnable)
+            {
+                if (datetime > DateTimeOffset.UtcNow)
+                {
+                    this.RunState = RunState.Ignored;
+                    string reason = (string)this.Properties.Get(PropertyNames.SkipReason);
+                    reason = string.Format("Ignoring until {0}. {1}", datetime.ToString("u"), reason);
```

Comment:
We need to deduplicate this code. I just realized that this is a copy of the code in IgnoreAttribute.

@nunit/framework-team Any thoughts on where we should keep common code like this?

Reply:
I agree, but this PR has been out there for many months. Other than this, I think it looks good and can be merged. Maybe we leave the refactoring to another PR?

Reply:
@rprouse That sounds good!